stdlib/test/collections/test_list.mojo

```diff
@@ -816,6 +816,18 @@ def test_list_contains():
     # assert_equal(List(0,1) in y,False)
 
 
+
+def test_list_eq_ne():
+    var l1 = List[Int](1, 2, 3)
+    var l2 = List[Int](1, 2, 3)
+    assert_true(l1 == l2)
+    assert_false(l1 != l2)
+
+    var l3 = List[Int](1, 2, 3, 4)
+    assert_false(l1 == l3)
+    assert_true(l1 != l3)
```

Comment:
Let me know what other use cases should I add here.

Reply:
Can you please add some tests with:
1. An empty list
2. Lists with memory-only types (such as `String`)

Reply:
Going to add those cases.